Customized samples 200720 (#133)
sendbird/SendBird-JavaScript: uikit-samples/README.md

## Sendbird UIKIT React

Find the docs at: https://docs.sendbird.com/javascript/ui_kit_getting_started

## Customized Samples

These samples show how to use and customize Sendbird UIKit. They are integrated in [codesandbox](https://codesandbox.io).<br />

> You can set your own APP_ID, USER_ID, and NICKNAME in `const.js` instead of already set things.<br />

Reviewer: ```suggestion
You have to set your own APP_ID, USER_ID, and NICKNAME in `const.js` in each example to make them work.<br />
```

And there are some commented links in the samples. They will be usefull for reference.

### 1-1) [Using SendBird App](https://codesandbox.io/s/1-1-using-sendbird-app-9xum5)

You can see how to use `App` component in this sample.

### 1-2) [Basic format of SendBird UIkit Customization](https://codesandbox.io/s/1-2-customization-basic-format-q4e6c)

> You can see a basic format for customizing in this sample.

Reviewer: origine -> origin

  Channel as SBConversation,
  ChannelList as SBChannelList,
  ChannelSettings as SBChannelSettings,
} from 'sendbird-uikit'
```

### 2-1) [Customizing MessageItem](https://codesandbox.io/s/2-1-customizing-messageitem-0nop5?file=)

You can see how to customize message item of the `Channel` component in this sample.

#### App.js
If you enable the toggle button, you can see that the message item customization is applied.

#### CustomizedApp.js
You can see a `renderChatItem` prop of the `Channel`(SBConversation) component.<br />
Set a function returning customized message item. You can get these function parameters.
```js
<Channel
  renderChatItem={({
    message,
    onDeleteMessage,
    onUpdateMessage,
    onResendMessage,
    emojiContainer,
  }) => {
    <CustomizedMessageItem />
  }}
>
```

#### CustomizedMessageItems
Make your own customized message item components here.

### 2-2) [Customizing MessageListParams](https://codesandbox.io/s/2-2-customizing-messagelistparams-45573)

You can see how to customize message list of the `Channel` component in this sample.

#### App.js
If you enable the toggle button, you can see that the message list customization are applied.<br />

> You will see the message items only you sent.

Reviewer: ```suggestion
You will see only the messages that you sent
```

#### CustomizedApp.js
You can see a `queries` prop of the `Channel`(SBConversation) component.

```js
<Channel
  queries={{
    messageListParams: {
      senderUserIds: [USER_ID],
      prevResultSize: 30,
      includeReplies: false,
      includeReactions: false
    }
  }}
>
```

> > Use object json type input, don't create sendbird message list params instance

Reviewer: 👍

You can see how to customize channel preview item of the `ChannelList` component in this sample.

#### App.js
If you enable the toggle button, you can create channel using GroupChannelParams.

#### CustomizedApp.js
You can see this prop.
```js
<ChannelList
  onBeforeCreateChannel={handleOnBeforeCreateChannel}
>
```
You can get array of selectredUsers as a function parameter.<br />
And you should return `GroupChannelParams` *instance* in the function.
```js
const handleOnBeforeCreateChannel = (selectedUsers) => {
  const channelParams = new sdk.GroupChannelParams();
    channelParams.addUserIds(selectedUsers);
    channelParams.name = "Hello SendBird!!";
    channelParams.overUrl = null;
    channelParams.coverImage = null;
    channelParams.customType = HIGHLIGHT;
    return channelParams;
}
```

GroupChannelParams
* [SendBird API Reference](https://sendbird.github.io/core-sdk-javascript/module-model_params_groupChannelParams-GroupChannelParams.html)
* [GitHub SendBird.d.ts](https://github.com/sendbird/SendBird-SDK-JavaScript/blob/master/SendBird.d.ts#L1448)

#### CustomizedChannelPreviewItem.js

> If the customType of chanenl is `highlight`, show background color with yellow.

Reviewer: ```suggestion
If the customType of channel is `highlight`, show background color with yellow.
```
